src/dune/action_exec.ml

```diff
@@ -7,7 +7,15 @@ type exec_context =
   ; purpose : Process.purpose
   }
 
-let exec_run ~ectx ~dir ~env ~stdout_to ~stderr_to ~stdin_from prog args =
+type exec_environment =
```

Comment:
Should we just move this type to `Process` and have `run` take it as an argument?

Reply:
Yeah, that sounds good. @staronj, does that fit with the other changes you're working on?

Reply:
Not sure, `Process` have different variations of `run` that take different set of arguments and the `run` itself have most of the relevant arguments as optional arguments.